src/lib/decorate-text.jsx

```diff
@@ -0,0 +1,30 @@
+const React = require('react'); // eslint-disable-line
+const reactStringReplace = require('react-string-replace');
+
+/**
+ * Helper method that replaces @mentions and #hashtags in plain text
+ * 
+ * @param  {string}   text     string to convert
+ * @return {string}            string with links for @mentions and #hashtags
+ */
+module.exports = text => {
+    let replacedText;
+    
+    // Match @-mentions
+    replacedText = reactStringReplace(text, /@(\w+)/g, (match, i) => (
+        <a
+            href={`/users/${match}`}
+            key={match + i}
+        >@{match}</a>
+    ));
+    
+    // Match hashtags
+    replacedText = reactStringReplace(replacedText, /#(\w+)/g, (match, i) => (
```

Comment:
We probably also want to include hyphens and underscores here too? At least hyphens.

Reply:
Actually this raises the question of what we want to allow in hashtags, and internationalization. In `#thebestever!` does the '!' get included in the hashtag. If we allow punctuation and symbols, is it any non-whitespace character, or specific ones. For non-latin characters there's a babel plugin to support Unidcode regex: https://github.com/babel/babel/blob/master/packages/babel-plugin-proposal-unicode-property-regex/README.md

For now let's just support words with underscore and hyphen. I'll open up an issue to discuss what else to include in hashtags. I think that we should at least support non-english characters in the long run - will that work with the search engine?